api/controllers/SearchController.js

```diff
@@ -1,11 +1,21 @@
 var findCommunityIds = Promise.method(req => {
   if (req.param('communityId')) {
     return [req.param('communityId')]
+  } else if (req.param('type') === 'communities' && req.param('moderated') && Admin.isSignedIn(req)) {
+    return Community.fetchAll()
+    .then(cs => Promise.map(cs.models, c => c.id))
   } else {
     return Promise.join(
       Network.activeCommunityIds(req.session.userId),
       Membership.activeCommunityIds(req.session.userId)
-    ).then(ids => _(ids).flatten().uniq().value())
+    ).then(dupIds => {
+      var ids = _(dupIds).flatten().uniq().value()
+      if (req.param('moderated')) {
+        return Promise.filter(ids, id => Membership.hasModeratorRole(req.session.userId, id))
```

Comment:
however, this is going to be a pretty slow operation for Ed, because it's one query per membership. what i would suggest instead is to create a new class method on Membership (or add a parameter to `activeCommunityIds`) that only returns the memberships with moderator roles.